net/udp/udp_input.c

```diff
@@ -243,7 +290,14 @@ static int udp_input(FAR struct net_driver_s *dev, unsigned int iplen)
         {
           /* We'll only get multiple conn when we support SO_REUSEADDR */
 
-#ifdef CONFIG_NET_SOCKOPTS
+#if defined(CONFIG_NET_SOCKOPTS) && defined(CONFIG_NET_BROADCAST)
+          /* Check if the destination address is a broad/multicast address */
+
+          if (!udp_is_broadcast(dev))
+            {
+              goto unicast;
```

Comment:
remove the goto unicast and move line 249-272 into if block

Reply:
Done.